pkg/controller/import-controller.go

```diff
@@ -139,15 +145,108 @@ func NewImportController(mgr manager.Manager, log logr.Logger, importerImage, pu
 	if err != nil {
 		return nil, err
 	}
-	if err := addImportControllerWatches(mgr, importController); err != nil {
+	if err := addImportControllerWatches(
+		mgr,
+		importController,
+		client,
+		log,
+	); err != nil {
 		return nil, err
 	}
 	return importController, nil
 }
 
-func addImportControllerWatches(mgr manager.Manager, importController controller.Controller) error {
+func addImportControllerWatches(mgr manager.Manager, importController controller.Controller, c client.Client, log logr.Logger) error {
 	// Setup watches
-	if err := importController.Watch(&source.Kind{Type: &corev1.PersistentVolumeClaim{}}, &handler.EnqueueRequestForObject{}); err != nil {
+
+	if err := importController.Watch(
+		&source.Kind{Type: &corev1.PersistentVolumeClaim{}},
+		handler.Funcs{
+			CreateFunc: func(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+				if evt.Meta == nil {
+					log.Error(nil, "CreateFunc received with no metadata", "event", evt)
+					return
+				}
+				q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+					Name:      evt.Meta.GetName(),
+					Namespace: evt.Meta.GetNamespace(),
+				}})
+			},
+			UpdateFunc: func(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+				if evt.MetaOld != nil {
+					q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+						Name:      evt.MetaOld.GetName(),
+						Namespace: evt.MetaOld.GetNamespace(),
+					}})
+				} else {
+					log.Error(nil, "UpdateEvent received with no old metadata", "event", evt)
+				}
+
+				if evt.MetaNew != nil {
+					q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+						Name:      evt.MetaNew.GetName(),
+						Namespace: evt.MetaNew.GetNamespace(),
+					}})
+				} else {
+					log.Error(nil, "UpdateEvent received with no new metadata", "event", evt)
+				}
+			},
+			DeleteFunc: func(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+				if evt.Meta == nil {
+					log.Error(nil, "DeleteEvent received with no metadata", "event", evt)
+					return
+				}
+				log.V(0).Info("----------------> DeleteEvent 1", "event", evt)
+
+				pvc := evt.Object.(*corev1.PersistentVolumeClaim)
+				owner := metav1.GetControllerOf(pvc)
+				// it has an owner pod, so it has to be a scratch PVC, so we need to fetch the controlling PVC
+				// PVC -> importerPod -> scratchPVC
+				if owner != nil && owner.Kind == "Pod" {
+					log.V(0).Info("---------------->  DeleteEvent 2 for PVC owner", "scratchPvc", pvc.Name, "owner", owner)
+
+					controllingPod := &corev1.Pod{}
+					if err := c.Get(context.TODO(), types.NamespacedName{Name: owner.Name, Namespace: evt.Meta.GetNamespace()}, controllingPod); err != nil {
+						// TODO: assume this is a scratch PVC without owner - so return an error, or schedule event for the pvc?
```

Comment:
So I wonder if we can leave it like it is right now (so we assume the pvc that has pod owner that has pvc owner must be scratch), or maybe we should mark scratch by Annotation and check the Annotation to be sure!

Reply:
For now we can assume that if the PVC has a pod owner, and that pod is owned by a different PVC, that it is a a scratch pvc.